nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/StandardHttpResponseMerger.java

```diff
@@ -0,0 +1,164 @@
+        endpointMergers.add(new ControllerServiceReferenceEndpointMerger());
+        endpointMergers.add(new ReportingTaskEndpointMerger());
+        endpointMergers.add(new ReportingTasksEndpointMerger());
+        endpointMergers.add(new DropRequestEndpiontMerger());
+        endpointMergers.add(new ListFlowFilesEndpointMerger());
+        endpointMergers.add(new ComponentStateEndpointMerger());
+        endpointMergers.add(new BulletinBoardEndpointMerger());
+        endpointMergers.add(new StatusHistoryEndpointMerger());
+        endpointMergers.add(new SystemDiagnosticsEndpointMerger());
+        endpointMergers.add(new CountersEndpointMerger());
+    }
+
+    @Override
+    public NodeResponse mergeResponses(final URI uri, final String httpMethod, final Set<NodeResponse> nodeResponses) {
+        final boolean hasSuccess = hasSuccessfulResponse(nodeResponses);
+        if (!hasSuccess) {
+            // It doesn't really matter which response we choose, as all are problematic.
+            final NodeResponse clientResponse = nodeResponses.iterator().next();
+
+            // Drain the response from all nodes except for the 'chosen one'. This ensures that we don't
+            // leave data lingering on the socket and ensures that we don't consume the content of the response
+            // that we intend to respond with
+            drainResponses(nodeResponses, clientResponse);
+            return clientResponse;
+        }
+
+        // Determine which responses are successful
+        final Set<NodeResponse> successResponses = nodeResponses.stream().filter(p -> p.is2xx()).collect(Collectors.toSet());
+        final Set<NodeResponse> problematicResponses = nodeResponses.stream().filter(p -> !p.is2xx()).collect(Collectors.toSet());
+
+        // Choose any of the successful responses to be the 'chosen one'.
+        final NodeResponse clientResponse = successResponses.iterator().next();
+
+        final EndpointResponseMerger merger = getEndpointResponseMerger(uri, httpMethod);
+        if (merger == null) {
+            return clientResponse;
+        }
+
+        return merger.merge(uri, httpMethod, successResponses, problematicResponses, clientResponse);
+    }
+
+    @Override
+    public Set<NodeResponse> getProblematicNodeResponses(final Set<NodeResponse> allResponses) {
+        // Check if there are any 2xx responses
+        final boolean containsSuccessfulResponse = hasSuccessfulResponse(allResponses);
+
+        if (containsSuccessfulResponse) {
+            // If there is a 2xx response, we consider a response to be problematic if it is not 2xx
+            return allResponses.stream().filter(p -> !p.is2xx()).collect(Collectors.toSet());
+        } else {
+            // If no node is successful, we consider a problematic response to be only those that are 5xx
+            return allResponses.stream().filter(p -> p.is5xx()).collect(Collectors.toSet());
+        }
```

Comment:
One of the benefits of functional programming is the ability to create a reference to a function and then reuse it. So the above could be further simplified by doing something like this:
```
Predicate<?> predicate = containsSuccessfulResponse ? p -> !p.is2xx() : p -> p.is5xx()
return allResponses.stream().filter(predicate).collect(Collectors.toSet());
```